### PR TITLE
feat: add C implementation for `math/base/special/log10`

### LINKED ISSUE #733: [RFC]: Add C implementation for `@stdlib/math/base/special/log10`
### Description

This RFC proposes adding C implementation for [`@stdlib/math/base/special/log10`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/@stdlib/math/base/special/log10).

```c
double stdlib_base_log10( const double x );
```

### Related Issues

None.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Pranavchiku**
Resolves #733.

## Checklist

- [x] update readme.md
- [x] include.gypi
- [x] binding.gyp
- [x] include/stdlib/math/base/special/
- [x] evalpoly.js
- [x] src
- [x] manifest.json
- [x] lib
- [x] examples
- [x] benchmark
- [x] test
* * *

@kgryte

### CODE REVIEW

**Comment by Pranavchiku** [comment]
@kgryte I am not sure about implementation in main.c, if you can have a look.

File: lib/node_modules/@stdlib/math/base/special/log10/benchmark/benchmark.native.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+// VARIABLES //
+
+var log10 = tryRequire( resolve( __dirname, './../lib/native.js' ) );
+var opts = {
+	'skip': ( log10 instanceof Error )
+};
+
+
+
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/log10/benchmark/benchmark.native.js]: I guess js linters are not working on my device, ( I actually had a fresh reinstallation of Ubuntu 22.04 )

**Comment by Pranavchiku** [comment]
All tests and benchmarks cleared, this PR can have a review.

File: lib/node_modules/@stdlib/math/base/special/log10/benchmark/benchmark.native.js (hunk 1 above)
Comment (kgryte) [reply lib/node_modules/@stdlib/math/base/special/log10/benchmark/benchmark.native.js]: They should work; however, you need to run `make init` on fresh repo installations (e.g., after cloning the project) to ensure that JS linting is setup correctly.

File: lib/node_modules/@stdlib/math/base/special/log10/README.md [hunk 2]
```diff
@@ -53,7 +53,7 @@ Evaluates the [common logarithm][common-logarithm].
 
 ```javascript
 var v = log10( 100.0 );
-// returns 2.0
+// returns ~2.0
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/log10/README.md]: Why would this be approximate?

From the REPL,

```
In [1]: base.log10(100)
Out[1]: 2
```

Comment (Pranavchiku) [reply lib/node_modules/@stdlib/math/base/special/log10/README.md]: Apologies for it, I don't actually remember why I did this, but yeah as it is not approximate so we don't need ~.

Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/log10/README.md]:
```suggestion
// returns 2.0
```

File: lib/node_modules/@stdlib/math/base/special/log10/benchmark/benchmark.native.js (hunk 1 above)
Comment (Pranavchiku) [reply lib/node_modules/@stdlib/math/base/special/log10/benchmark/benchmark.native.js]: Yeah, now it works well :)

**Review by Pranavchiku — COMMENTED** [review]
@kgryte I think this PR can have a final go through and this will then be ready to merge.

**Review by Planeshifter — APPROVED** [review]
LGTM!

File: lib/node_modules/@stdlib/math/base/special/log10/src/main.c [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+
+/* End auto-generated functions. */
+
+/**
+* Return `log(x) - (x-1)` for `x` in `~[sqrt(2)/2, sqrt(2)]`.
+*
+* @param {number} x - input value
+* @returns {number} function value
+*/
+double klog( const double x ) {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/log10/src/main.c]:
```suggestion
static double klog( const double x ) {
```

File: lib/node_modules/@stdlib/math/base/special/log10/src/main.c [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+
+/**
+* Evaluates the common logarithm (base ten).
+*
+* @param x    input value
+* @return     output value
+*
+* @example
+* double out = stdlib_base_log10( 4.0 );
+* // returns 0.6020599913279624
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/log10/src/main.c]:
```suggestion
* // returns ~0.602
```

**Review by kgryte — APPROVED** [review]
LGTM.

**Comment by kgryte** [comment]
@gunjjoshi Now that this is merged, would you be willing to migrate to `msun` 12.x?

**Comment by gunjjoshi** [comment]
Yes @kgryte, I'll migrate the C implementation for this to [`msun 12.2.0`](https://svnweb.freebsd.org/base/release/12.2.0/lib/msun/src/e_log10.c?revision=367086&view=markup) now.

**Comment by gunjjoshi** [comment]
Made a PR for this at #2215.

**Pull request merged**